### PR TITLE
Set up common rules for the codebase

### PR DESCRIPTION
**Pull request by 123mpozzi**
Set up common rules to ensure a pretty and unique codebase among maintainers

## Problem
Everybody has their preferences on code style and conventions and this is a problem when working on the same codebase. 
It is therefore necessary to establish common rules to agree on.  

This PR aims at adding tooling to create these rules, but **note** that the actual rules to be applied have not yet been agreed upon and can still be discussed.  
For example, a very popular style is the [Airbnb one](https://github.com/airbnb/javascript).

## Changes
- add **eslint** as the linter to ensure good practices and same code conventions
  Can be manually run with: `npm run lint`
- add **editorconfig** to setup common indentation and line-endings rules
- add **prettier** as the code formatter to ensure the same codestyle
  Can be manually run with: `npm run format`
- add git pre-commit hooks using **husky** to enforce the previous tooling automatically

## Ignore git hooks
In case you want to just fast commit and ignore the hooks, just use the following flag:
`git commit --no-verify`

### CODE REVIEW

**Review by SimoneDevkt — COMMENTED** [review]
event 'load' is now useless, you can delete it

File: src/main.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ const idCanvas = 'gameCanvas'
 document.querySelector<HTMLDivElement>('#app')!.innerHTML = `<canvas id="${idCanvas}"></canvas>`
 
 window.addEventListener('load', function() {
```
Comment (SimoneDevkt) [inline src/main.ts]: load eventlistener is now useless, remove it

**Review by SimoneDevkt — COMMENTED** [review]
ok

File: src/main.ts (hunk 1 above)
Comment (123mpozzi) [reply src/main.ts]: Address in 7940acb65e6c8d11f23e279dd3143786db944909

**Pull request merged**